charts/redpanda/ci/09-initcontainers-resources-values.yaml

```diff
@@ -80,10 +80,10 @@ statefulset:
           - |
             set -xe
             echo "Hello World!"
-  extraVolumes: |-
+  extraVolumes:
     - name: test-extra-volume
       secret:
-        secretName: {{ (include "redpanda.fullname" . | trunc 50 ) }}-sts-lifecycle
+        secretName: redpanda-sts-lifecycle
```

Comment:
That change broke chart testing. 

All Redpanda containters can not mount `redpanda-sts-lifecycle` volume:
```
2024-07-07T17:55:33.8977130Z 103s        Warning   FailedMount             pod/redpanda-a7e5e263hu-1                                                                           kubelet, chart-testing-worker3                                                                     MountVolume.SetUp failed for volume "test-extra-volume" : secret "redpanda-sts-lifecycle" not found                                                                                                                                             9m57s        12      redpanda-a7e5e263hu-1.17dfff5833e88513
2024-07-07T17:55:33.8980303Z 103s        Warning   FailedMount             pod/redpanda-a7e5e263hu-0                                                                           kubelet, chart-testing-worker                                                                      MountVolume.SetUp failed for volume "test-extra-volume" : secret "redpanda-sts-lifecycle" not found                                                                                                                                             9m57s        12      redpanda-a7e5e263hu-0.17dfff582c2e8e89
2024-07-07T17:55:33.8983033Z 102s        Warning   FailedMount             pod/redpanda-a7e5e263hu-2                                                                           kubelet, chart-testing-worker2                                                                     MountVolume.SetUp failed for volume "test-extra-volume" : secret "redpanda-sts-lifecycle" not found                                                                                                                                             9m56s        12      redpanda-a7e5e263hu-2.17dfff586d611153
```

I wouldn't invest in chart testing shenanigans. My suggestion is to remove 
https://github.com/redpanda-data/helm-charts/blob/d2444af416d0e182f476bb231f90d7b2b7c636d7/charts/redpanda/ci/09-initcontainers-resources-values.yaml#L80-L87

The problem comes from the fact that `redpanda.fullname` could be different every time chart testing executes as chart release name is generated e.g. `redpanda-uf00cn755x`.
https://github.com/redpanda-data/helm-charts/actions/runs/9811061785/job/27133852352?pr=1408#step:18:21041

Reply:
Yeah, I'm going to update the unit test here. I think the utility of this templating is somewhat reduced in practice.